backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/projection/impl/LuceneFieldHighlightProjection.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.search.projection.impl;
+
+import java.util.List;
+import java.util.Set;
+
+import org.hibernate.search.backend.lucene.lowlevel.collector.impl.Values;
+import org.hibernate.search.backend.lucene.search.common.impl.LuceneSearchIndexScope;
+import org.hibernate.search.backend.lucene.search.common.impl.LuceneSearchIndexValueFieldContext;
+import org.hibernate.search.backend.lucene.search.highlighter.impl.LuceneSearchHighlighter;
+import org.hibernate.search.engine.search.common.spi.SearchQueryElementFactory;
+import org.hibernate.search.engine.search.loading.spi.LoadingResult;
+import org.hibernate.search.engine.search.projection.dsl.spi.HighlightProjectionBuilder;
+
+import org.apache.lucene.analysis.Analyzer;
+
+public class LuceneFieldHighlightProjection implements LuceneSearchProjection<List<String>> {
+
+	private final Set<String> indexNames;
+	private final Analyzer analyzer;
+	private final String absoluteFieldPath;
+	private final String highlighterName;
+
+	private LuceneFieldHighlightProjection(Builder builder) {
+		this( builder.scope, builder.field, builder.highlighterName() );
+	}
+
+	LuceneFieldHighlightProjection(LuceneSearchIndexScope<?> scope,
+			LuceneSearchIndexValueFieldContext<?> field,
+			String highlighterName) {
+		this.indexNames = scope.hibernateSearchIndexNames();
+		this.analyzer = field.type().searchAnalyzerOrNormalizer();
+		this.absoluteFieldPath = field.absolutePath();
+		this.highlighterName = highlighterName;
+	}
+
+	@Override
+	public String toString() {
+		return getClass().getSimpleName() + "["
+				+ "absoluteFieldPath=" + absoluteFieldPath
+				+ "highlighterConfigurationName=" + highlighterName
+				+ "]";
+	}
+
+	@Override
+	public Set<String> indexNames() {
+		return indexNames;
+	}
+
+	@Override
+	public FieldHighlightExtractor request(ProjectionRequestContext context) {
+		context.requireAllStoredFields();
```

Comment:
I've changed the plain highlighter to be more similar to the other two since those were using `LeafReaderContext` to access the required data for highlighting. Hence this part `context.requireAllStoredFields();` is not needed anymore. 
As for the nesting, I think that should work now for the Lucene backend. As for the Elasticsearch one ... It doesn't seem like the nesting structure can be preserved in the `highlight` response; as a result, all the fields are accumulated. Because of that, you'll notice quite a few `assume***(..)` in `HighlightProjectionBaseIT`.